### PR TITLE
Add issue template for bug reports

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
This PR adds an issue template form to collect info needed to triage bug reports.

The form asks the following questions;
1. What happened?
2. What did you expect to happen?
3. What version of Unity are you running?
4. Which platform are you having an issue with?
5. Relevant log output
6. Code of conduct

If there are any other questions I should ask, feel free to drop a comment so I can incorporate the changes. Some things I'm not sure about are
1.  Which versions of Unity do we want to include in the dropdown?
2. Do we want to ask for specific version of iOS/Android
3. Should we ask which libraries are being used 

**[📝 Rendered form](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/484a65715435392e77cf5c7642e65940a83778ab/.github/ISSUE_TEMPLATE/BUG_REPORT.yml)**

###### References
* [Creating issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms)
* [Creating issue templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-templates)
* [Configuring the template chooser](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser)

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/454)

### CODE REVIEW

**Comment by iAmWillShepherd** [comment]
@adamschlesinger, I think your input on this is needed. Are there any other pieces of information you'd like to have?

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      label: Unity version
+      description: What version of Unity are you running?
+      options:
+        - Unity 2021.2.12 (latest)
+        - Unity 2021.x
+        - Unity 2020.x
+        - Unity 2019.x
+    validations:
+      required: true
+  - type: dropdown
```
Comment (jkasten2) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:42]: Could this be a check box so they can select both? If not can we add another options for "Android & iOS"?

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - type: textarea
+    id: what-are-expectations
+    attributes:
+      label: What did you expect to happen?
+      description: Also tell us, what did you expect to happen?
+      placeholder: Tell us what you see!
+      value: "A bug happened!"
+    validations:
+      required: true
+  - type: dropdown
```
Comment (jkasten2) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:30]: Is it possible provide an other option so they can fill it in?

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    id: unity-version
+    attributes:
+      label: Unity version
+      description: What version of Unity are you running?
+      options:
+        - Unity 2021.2.12 (latest)
```
Comment (jkasten2) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:36]: Would be better if this was just "Latest", or even better if we can let them enter in a custom value.

**Review by jkasten2 — CHANGES_REQUESTED** [review]
Don't know the format very well for these so most of the feedback is questions.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 2 above)
Comment (iAmWillShepherd) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml:30]: Logic is limited, so we could add an optional text box for other info. I only need to add a description to make it clear the field is to be used if other is selected.

**Comment by iAmWillShepherd** [comment]
@jkasten2 It's not currently possible to make sure at least one checkbox is selected ([see GitHub form schema: checkboxes](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#checkboxes)).

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 3 above)
Comment (iAmWillShepherd) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: Fixed in https://github.com/OneSignal/OneSignal-Unity-SDK/pull/454/commits/cdf8558976122cc7e54be44835d4cadb9c2d0703.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 1 above)
Comment (iAmWillShepherd) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: Fixed in https://github.com/OneSignal/OneSignal-Unity-SDK/pull/454/commits/bd9d3ce5615eb6037fd07d0048fff1291f77a6f6.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 2 above)
Comment (iAmWillShepherd) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: I went with the textbox option.

**Comment by jkasten2** [comment]
> @jkasten2 It's not currently possible to make sure at least one checkbox is selected ([see GitHub form schema: checkboxes](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#checkboxes)).

O I see the `dropdown` type has `multiple` it looks like we could use instead.

**Review by adamschlesinger — CHANGES_REQUESTED** [review]
We need a "What version of the OneSignal Unity SDK are you using?" section

> *[Reviewable](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/454#-:-Mwgf5i52qAOgVzUgGzl:b2gawci)* status: 0 of 1 files reviewed, 5 unresolved discussions (waiting on @iAmWillShepherd and @jkasten2)

___
*[.github/ISSUE_TEMPLATE/BUG_REPORT.yml, line 16 at r3](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/454#-Mwgejsb8vxBf227ZOd6:-Mwgejsb8vxBf227ZOd7:b-xvopwl) ([raw file](https://github.com/onesignal/onesignal-unity-sdk/blob/df165ff7558ae05e392591a44c0113e61c2f392a/.github/ISSUE_TEMPLATE/BUG_REPORT.yml#L16)):*
> ```YAML
>     attributes:
>       label: What happened?
>       description: Also tell us, what steps are required to reproduce.
> ```

Can "Steps to Reproduce" be a dedicated section?
___
*[.github/ISSUE_TEMPLATE/BUG_REPORT.yml, line 45 at r3](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/454#-MwgeNIjAfEmRt9zWdFQ:-MwgeNIjAfEmRt9zWdFR:byvu8gv) ([raw file](https://github.com/onesignal/onesignal-unity-sdk/blob/df165ff7558ae05e392591a44c0113e61c2f392a/.github/ISSUE_TEMPLATE/BUG_REPORT.yml#L45)):*
> ```YAML
>       description: Which platform are you having an issue with?
>       multiple: true
>       options:
> ```

I'd like to have an option for "Other" as it's unfortunately feasible we could break unsupported platforms.

**Comment by iAmWillShepherd** [comment]
> We need a "What version of the OneSignal Unity SDK are you using?" section

@adamschlesinger  That's already there. It's labeled `unity-version.` Perhaps you were expecting a dropdown?

**Review by jkasten2 — APPROVED** [review]

**Review by adamschlesinger — COMMENTED** [review]
Yes, we have "Unity version" but we also need "OneSignal Unity SDK version". ie:
```
              Unity version: 2018.4.34f1
OneSignal Unity SDK version: 3.0.0-beta.5
```

> *[Reviewable](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/454#-:-Mx0fKMnEgpyUdfAszDf:b5ba3oo)* status: 0 of 1 files reviewed, 5 unresolved discussions (waiting on @adamschlesinger, @iAmWillShepherd, and @jkasten2)

**Comment by iAmWillShepherd** [comment]
@adamschlesinger I added the new field for SDK version along with other minor edits.

**Review by adamschlesinger — APPROVED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/454)* status: all files reviewed, 3 unresolved discussions (waiting on @jkasten2)

**Review by jkasten2 — APPROVED** [review]

**Pull request merged**